lib/screen/customer_support/support_thread_page.dart

```diff
@@ -328,13 +447,31 @@ class _SupportThreadPageState extends State<SupportThreadPage> {
 
     if (mounted) {
       setState(() {
+        String lastMessage = "";
+        if (issueDetails.messages.isNotEmpty){
+          lastMessage = issueDetails.messages[0].message ?? "";
+        }
+
         _status = issueDetails.issue.status;
+        _isRated = issueDetails.issue.rating > 0 &&
+            issueDetails.issue.status == "closed" &&
+            (lastMessage.contains(RATING_MESSAGE_START) ||
+            lastMessage.contains(STAR_RATING));
         _reportIssueType = issueDetails.issue.reportIssueType;
         _messages = parsedMessages;
       });
     }
   }
 
+  bool _isRating(types.Message message) {
+    List<int> listStar = [1, 2, 3, 4, 5];
+    if (message.metadata?["rating"] != null &&
```

Comment:
too many repeated `message.metadata?["rating"]`, can we convert to a short variable to easier to handle?

Reply:
this message object is from 'package:flutter_chat_types/flutter_chat_types.dart', we shall not change that class. Could you suggest any way to go around this, or i will leave it that way.

Reply:
I mean define local a variable ex `final rating = message.metadata?["rating"]`, and check this variable instead.